### PR TITLE
[Tests-Only] Move unauthenticated tests to jest

### PR DESCRIPTION
**Pull request by saw-jan**
Move unauthenticated tests to jest
 - appsTest.js
 - capabilitiesTest.js
 - filesTest.js
 - groupTest.js
 - sharingTest.js
 - userTest.js

part of #645 & #646

### CODE REVIEW

**Review by dpakach — COMMENTED** [review]
LGTM :+1:

File: src/helperFunctions.js [hunk 1]
```diff
@@ -295,9 +295,6 @@ class helpers {
       err = 'Please specify an authorization first.'
     }
 
-    if (err) {
-      throw new Error(err)
-    }
```
Comment (individual-it) [inline src/helperFunctions.js:300]: why deleting this?

**Review by individual-it — CHANGES_REQUESTED** [review]

File: src/helperFunctions.js (hunk 1 above)
Comment (saw-jan) [reply src/helperFunctions.js:300]: @dipak has better explanation.
IMO, Promise will reject if there is any error i.e. `err` and that `if` block was blocking any GET requests made without auth-header which were needed for unauthenticated tests.

Comment (individual-it) [reply src/helperFunctions.js:300]: it was added here https://github.com/owncloud/owncloud-sdk/pull/637/files#diff-b58c3c88931c8c80492c58e8b2f7f2000efc13c8eb29120bbdcc033f3f815873R298-R300 and is not needed

**Review by individual-it — APPROVED** [review]

**Pull request merged**